### DISCUSSION TITLE
Allow to encode uncertainty in pname

### DISCUSSION THREAD

**Discussion by martha-thomae** [music-encoding/music-encoding#809] [Q&A] [answered]
Recently, a colleague of mine found out that sometimes (as in the case of Andrew Hughes encodings of chants) you need to encode the fact that you are not certain about the pitch of a note. For that particular project, my colleague is using `pname="?"`. However, this is not valid MEI (pname's value is restrcited to characters `a-g`). 

Is there a way to encode the uncertainty of the pitch? Should we consider adding other values to `pname`?

**Comment by davidrizo** [comment]
Hi Martha,

There is an attribute @cert in Editorial Markup (https://music-encoding.org/guidelines/v4/content/scholarlyediting.html#editTrans <https://music-encoding.org/guidelines/v4/content/scholarlyediting.html#editTrans>) that may be useful for it

Best,
David

**Comment by rettinghaus** [comment]
Better way would be to use `unclear` element:
```xml
<unclear>
   <note />
</unclear>
```

if there are multiple possibilities wrap it in `choice`:
```xml
<choice>
   <unclear>
      <note />
   </unclear>
   <unclear>
      <note />
   </unclear>
</choice>
```

And on `unclear` you have the `@cert` attribute available:
```xml
<choice>
   <unclear cert="high">
      <note />
   </unclear>
   <unclear cert="low">
      <note />
   </unclear>
</choice>
```

**Comment by rettinghaus** [comment]
see Guidelines https://music-encoding.org/guidelines/v4/content/scholarlyediting.html#edittransOmission

**Comment by martha-thomae** [comment]
Thank you so much, @rettinghaus and @davidrizo!
I will pass this information along and refer my colleague to this issue if he wants to ask more questions. Thank you! :)

**Comment by musicEnfanthen** [comment]
Hi all, thanks for the interesting discussion.

@rettinghaus Three questions arise for me:
1) `unclear` would declare the entire note element as unclear (i.e., it is unclear if the entity we describe is a note at all). In the given case, however, only the pname attribute is unclear. Is there something like the pattern described in [chapter 11.2.4.1](https://music-encoding.org/guidelines/v4/content/scholarlyediting.html#edittransOmission) for `artic` also for the pname (and other) attributes?

```
<note dur="4" oct="4" pname="c">
   <unclear reason="ink blot">
      <artic artic="acc"/>
   </unclear>
</note>
```

2) Ignoring question 1 for now, the [choice pattern given above](https://github.com/music-encoding/music-encoding/issues/800#issuecomment-846103881) will work quite well for use cases, when a pname is not quite clear to identify, but there is a rough guess (e.g. it could be `a` or `g`). But what to do when there is no guess at all... Would you need to put all pnames from `a` to `g` (including inflections) into a choice of `unclear`s?

3) What does `@cert` refer to when used with `unclear`? Is it the certainty level of the feature/element enclosed by `unclear`, or is it the `unclear` element itself? If the latter, `<unclear cert="high">` would mean that it is "highly unclear", whereas the former would mean that the encoder is highly certain about the enclosed feature/element.
In chapter 11.2.2, there is an example for cert attribute with corr element, described in the text as
> The att.edit class makes available a @cert attribute (through att.evidence), which may be used to indicate the degree of editorial confidence in a particular correction
```
<corr cert="high">
         <note dur="4" oct="4" pname="g"/>
</corr>
```
If transferred to `unclear`, it would suggest that the level of unclearness is very high in `<unclear cert="high">`.

**Comment by rettinghaus** [comment]
@musicEnfanthen to 1 and 2: there's nothing for individual attributes, as far as I know. But let me ask you, would a note without a position relative to a staff (described by `@loc` or `@pname`/`@oct`) be in all cases a note? It would give you a hard time to encode all blobs and strokes on a manuscript as noteheads and stems.

**Comment by ahankinson** [comment, marked as answer]
When encoding different readings of a particular note it is common practice to encode the whole note element, even if part of it is different:

```
<app>
  <rdg>
      <note pname="c" />
  </rdg>
  <rdg>
      <note pname="c" accid="s" />
  </rdg>
</app>
```

Most of the examples in the Scholarly Editing chapter of the guidelines follow this pattern. 

When no pitch is discernable, it is OK to have a `<note>` element with no `@pname` -- This is the recommended practice for unheighted neume notation, for example, where the pitch is not absolute, but we know we have a "pitch event" -- that is, something is sounded, but we can't tell you what the exact note is.

Remember that in MEI the `<note>` is not simply the visible symbol! It is also (primarily?) the logical function of a note, which is to say that there is a musical event happening in a stream of other events -- what it looks like on the page is somewhat immaterial. You may be uncertain about particular attributes of that event (like its pitch or accidental) but the fact that *something* musical is happening isn't in doubt. You could also alternate the `<note>` and `<rest>` elements if you are uncertain about whether something is a sounding or non-sounding event.

It is also helpful to consult the [TEI Guidelines](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-unclear.html) for the elements in MEI that are borrowed from there. For example, you could encode this particular case as:

```
<unclear reason="illegible">
     <note />
</unclear>
```

**Comment by martha-thomae** [comment]
Thank you, @ahankinson!

**Comment by ahankinson** [comment]
@martha-thomae can this issue be closed now?

**Comment by ahankinson** [comment]
Since this is more of a support discussion rather than a feature request (right now) I am going to move this to the "Discussions" tab.

**Reply by martha-thomae** [reply]
Thanks for moving this here. I will start using the "Discussions" tab for technical support from now on. Thanks!

**Comment by ahankinson** [comment]
@martha-thomae if you’re satisfied with the answers here could you mark this discussion as “answered”? Thanks!

**Reply by martha-thomae** [reply]
Yes. Thank you, Andrew! :)
